City-of-Helsinki/infofinland-ui
# UHF-10876

# [UHF-10876](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10876)

The functionality cannot fully be tested on local since adding localhost:3000 to askem-backend as allowed url might be bad idea.

Added askem and removed the original contact form. [Askem doc is here](https://docs.askem.com/#basic-embed-code)

Currently the div with 'askem'-class is only added to article page layout. 
The library is loaded on first page load and the reset-function (added by askem-library) is called on every  page load.

## How to install

- Add RNS api key to local.env, for example `REACT_AND_SHARE_APIKEY_FI=does-not-matter_wont-work-on-local-anyway`

- Run `make fresh` on backend
- Run `nvm use; npm install -g yarn`
- Run `yarn build; yarn start` (prefetching is disabled on `yarn run dev`)

## How to test
At this point we must rely on the errors to see when the block would render

- Open console from browser to see when the errors trigger.
- Open front page on finnish
  - You should not find div with `class="askem` from the page.
- Click any other link on sidebar menu 
  - You should see the div with 'askem' class in the dom  
  - If you have only finnish rns-key added, you should see:
    - Akem-error on inspector on first page load on any page
    - Askem-error on inspector on page change from one finnish page to another
    - No Askem-error on page change from finnish to english

## Other pull requests

https://github.com/City-of-Helsinki/drupal-infofinland/pull/384




[UHF-10876]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10876?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
